Build/CIRCLECI_SETUP.md

```diff
@@ -0,0 +1,246 @@
+   the CircleCI settings.
+    
+    >For jilleJr/Newtonsoft.Json-for-Unity.Converters, that page is found at
+    ><https://app.circleci.com/settings/project/github/jilleJr/Newtonsoft.Json-for-Unity.Converters/environment-variables>
+
+    Here's a table of the desired variables:
+
+    | File                                   | Variable name                         |
+    | -------------------------------------- | ------------------------------------- |
+    | Unity_v2018.4.14f1.ulf.b64             | UNITY_2018_4_14f1_LICENSE_CONTENT_B64 |
+    | Unity_v2019.2.11f1.ulf.b64             | UNITY_2019_2_11f1_LICENSE_CONTENT_B64 |
+    | Unity_v2020.1.0b6-linux-il2cpp.ulf.b64 | UNITY_2020_1_0b6_LICENSE_CONTENT_B64  |
+
+3. All done!
+
+If you're curious where the variable names are references; search for them in
+the `.circleci/config.yml` file.
+
+## NPM auth token
+
+Whatever NPM registry used, it needs an authentication token for pushing the
+new package version.
+
+### NPM auth for Cloudsmith
+
+1. Go to Cloudsmith API settings <https://cloudsmith.io/user/settings/api/>
+
+2. If using a different NPM registry, update the `deploy-cloudsmith` job 
+   inside the `.circleci/config.yml` file to use the appropriate URLS.
+
+3. Copy the content of your Cloudsmith API token and add it as variable
+   `NPM_AUTH_TOKEN` inside the CircleCI settings.
+
+    >For jilleJr/Newtonsoft.Json-for-Unity.Converters, that page is found at
+    ><https://app.circleci.com/settings/project/github/jilleJr/Newtonsoft.Json-for-Unity.Converters/environment-variables>
+
+4. All done!
+
+## Git deploy
+
+Deployment is also done via GitHub. Deployment to GitHub is just done by
+pushing the changes to the `upm` branch and then tagging that version with
+the version number.
+
+To be able to deploy to your repo, the build uses an SSH key for write access
+and optionally a GPG key for commit/tag signing.
+
+### UPM branch
+
+Ensure you have a branch called `upm`
+
+### Git SSH key
+
+1. Generate a new SSH key, replacing `your_email@example.com` with
+   the email address of the GitHub account that will be used for the auto
+   deployment.
+
+    ```sh
+    $ ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+    ```
+
+    - When you're prompted to "Enter a file in which to save the key,", enter a
+      new location such as `/home/your_user/.ssh/newtonsoft.json-for-unity.converters_rsa`
+
+    - At the prompt, **enter no passphrase**.
+
+2. Copy the content of the public key file
+   (ex: `/home/your_user/.ssh/newtonsoft.json-for-unity.converters_rsa.pub`)
+   into the GitHub project as a deploy key, naming it something like
+   "Newtonsoft.Json-for-Unity.Converters UPM branch deployment key"
+
+    >For jilleJr/Newtonsoft.Json-for-Unity.Converters, that page is found at
+    ><https://github.com/jilleJr/Newtonsoft.Json-for-Unity.Converters/settings/keys/new>
+
+    - Make sure to check the "Allow write access" checkbox.
+
+3. Copy the content of the private key file
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [[list-item-indent] Incorrect list-item indent: add 1 space](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity.Converters/pullRequest?prid=5461394)